jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/AbstractDiagnosticsCollector.java

```diff
@@ -0,0 +1,301 @@
+					return true;
+				}
+			} else if (importDeclaration.getElementName().equals(javaElementFQName)) {
+				return true;
+			}
+		}
+		return false;
+    }
+    
+	/**
+	 * Returns true if the given Java class imports one of the given Java elements and
+	 * false otherwise.
+	 *
+	 * @param type    	 			Java class.
+	 * @param javaElementFQName		given Java element fully qualified names.
+	 * @return true if the Java class imports one of the given Java elements and false otherwise.
+	 */
+    protected static boolean isImportedJavaElement(ICompilationUnit unit, String[] javaElementFQNames) throws JavaModelException {
+		IImportContainer container = unit.getImportContainer();
+		if (container == null) {
+			return false;
+		}
+		
+		// The following code uses JDT internal class and looks like
+		// ICompilationUnit#getImports()
+		// To avoid creating an array of IImportDeclaration, we do the following code:
+		JavaModelManager manager = JavaModelManager.getJavaModelManager();
+		Object info = manager.getInfo(container);
+		if (info == null) {
+			if (manager.getInfo(unit) != null) {
+				// CU was opened, but no import container, then no imports
+				// return NO_IMPORTS;
+				return false;
+			} else {
+				try {
+					unit.open(null);
+				} catch (JavaModelException e) {
+					e.printStackTrace();
+				} // force opening of CU
+				info = manager.getInfo(container);
+				if (info == null)
+					// after opening, if no import container, then no imports
+					// return NO_IMPORTS;
+					return false;
+			}
+		}
+		IJavaElement[] elements = ((ImportContainerInfo) info).getChildren();
+		for (IJavaElement child : elements) {
+			IImportDeclaration importDeclaration = (IImportDeclaration) child;
+			if (importDeclaration.isOnDemand()) {
+				String fqn = importDeclaration.getElementName();
+				String qualifier = fqn.substring(0, fqn.lastIndexOf('.'));
+				boolean imports = Stream.of(javaElementFQNames).anyMatch(elementFQName -> {
+					return qualifier.equals(elementFQName.substring(0, elementFQName.lastIndexOf('.')));
+				});				
+				if (imports == true) {
+					return true;
+				}
+			} else {
+				String importName = importDeclaration.getElementName(); 
+				if (Stream.of(javaElementFQNames).anyMatch(elementFQName -> importName.equals(elementFQName)) == true)
+					return true;
+			}
+		}
+		return false;
+    }
+    
+	/**
+	 * Returns true if the given Java class implements one of the given interfaces and
+	 * false otherwise.
+	 *
+	 * @param type    	 		Java class.
+	 * @param interfaceFQNames 	given interfaces with fully qualified name.
+	 * @return true if the Java class implements one of the given interfaces and false otherwise.
+	 */
+    protected static boolean doesImplementInterfaces(IType type, String[] interfaceFQNames) throws JavaModelException {
+    	String[] interfaceNames = type.getSuperInterfaceNames();
+
+    	// should check import statements first for the performance?

```

Comment:
is this a TODO?

Reply:
I just put a question there, not sure how expensive the following API is.

Reply:
Let's open a follow up investigative issue to track this as an enhancement